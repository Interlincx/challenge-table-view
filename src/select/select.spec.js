import React from 'react'
import Select from './select'
import { MenuItem } from '@material-ui/core'

const minProps = {
  items: ['United States', 'Canada', 'Mexico'],
  value: ''
}

test('renders a Select', () => {
  const wrapper = shallow(<Select {...minProps} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render 3 menu items', () => {
  const wrapper = shallow(<Select {...minProps} />)
  wrapper.simulate('click')
  expect(wrapper.find(MenuItem)).toHaveLength(3)
})
