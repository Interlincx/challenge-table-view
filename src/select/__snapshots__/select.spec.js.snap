// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Select 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <WithStyles(InputLabel)
    style={
      Object {
        "fontSize": 12,
      }
    }
  />
  <WithStyles(Select)
    fullWidth={true}
    value=""
  >
    <WithStyles(MenuItem)
      key="United States"
      value="United States"
    >
      United States
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="Canada"
      value="Canada"
    >
      Canada
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="Mexico"
      value="Mexico"
    >
      Mexico
    </WithStyles(MenuItem)>
  </WithStyles(Select)>
</div>
`;
